scipy/stats/biasedurn.pyx

```diff
@@ -0,0 +1,79 @@
+# distutils: language = c++
+
+from .BiasedUrn cimport CFishersNCHypergeometric, StochasticLib3
+import numpy as np
+
+cdef class _PyFishersNCHypergeometric:
+    cdef CFishersNCHypergeometric c_fnch
+
+    def __cinit__(self, int n, int m, int N, double odds, double accuracy):
+        self.c_fnch = CFishersNCHypergeometric(n, m, N, odds, accuracy)
+
+    def mode(self):
+        return self.c_fnch.mode()
+
+    def mean(self):
+        return self.c_fnch.mean()
+
+    def variance(self):
+        return self.c_fnch.variance()
+
+    def probability(self, int x):
+        return self.c_fnch.probability(x)
+
+    def moments(self):
+        cdef double mean, var
+        self.c_fnch.moments(&mean, &var)
+        return mean, var
+
+cdef class _PyStochasticLib3:
+    cdef StochasticLib3 c_sl3
+
+    def __cinit__(self, int seed):
+        self.c_sl3 = StochasticLib3(seed);
+        
+    def FillCache(self, double[::1] rand_cache):
+        self.c_sl3.FillCache(&rand_cache[0], rand_cache.shape[0]);
+        
+    def Random(self):
+        x = self.c_sl3.Random();
+        return x;
+        
+    def SetAccuracy(self, double accur):
+        return self.c_sl3.SetAccuracy(accur)
+        
+    def rvs_fisher(self, int n, int m, int N, double odds, int size, random_state=None, int FOS=2):
```

Comment:
if there won't be any dtype differences here (which I don't see how since you don't specify it) then we can accelerate this part with full cythonization via typing all vars and rvs array. Also if the arrays are big `xrange` is better for Cython instead of `range`.